test_runner/regress/test_lsn_mapping.py

```diff
@@ -108,6 +111,64 @@ def test_lsn_mapping(neon_env_builder: NeonEnvBuilder):
         assert Lsn(result["lsn"]) >= last_flush_lsn
 
 
+def test_get_lsn_by_timestamp_cancelled(neon_env_builder: NeonEnvBuilder):
+    """
+    Test if cancelled pageserver get_lsn_by_timestamp request is correctly handled.
+    Added as an effort to improve error handling and avoid full anyhow backtrace.
+    """
+
+    env = neon_env_builder.init_start()
+    env.pageserver.allowed_errors.extend(
+        [
+            ".*request was dropped before completing.*",
+            ".*Cancelled request finished with an error: Cancelled\n",
+        ]
+    )
+
+    client = env.pageserver.http_client()
+    failpoint = "find-lsn-for-timestamp-pausable"
+    client.configure_failpoints((failpoint, "pause"))
+
+    with env.endpoints.create_start("main") as endpoint:
+        conn = endpoint.connect()
+        cur = conn.cursor()
+        cur.execute("CREATE TABLE foo (x integer)")
+        cur.execute("INSERT INTO foo VALUES(0)")
+        # Get the timestamp at UTC
+        probe_timestamp = query_scalar(cur, "SELECT clock_timestamp()").replace(tzinfo=None)
+        cur.execute("INSERT INTO foo VALUES (-1)")
+
+        with ThreadPoolExecutor(max_workers=1) as exec:
+            # Request get_lsn_by_timestamp, hit the pausable failpoint
+            failing = exec.submit(
+                client.timeline_get_lsn_by_timestamp,
+                env.initial_tenant,
+                env.initial_timeline,
+                probe_timestamp,
+                timeout=2,
+            )
+
+            _, offset = wait_until(
+                20, 0.5, lambda: env.pageserver.assert_log_contains(f"at failpoint {failpoint}")
+            )
+
+            with pytest.raises(ReadTimeout):
+                failing.result()
+
+            client.configure_failpoints((failpoint, "off"))
+
+            msg, offset = wait_until(
+                20,
+                0.5,
+                lambda: env.pageserver.assert_log_contains(
+                    "Cancelled request finished with an error: Cancelled", offset
+                ),
+            )
+
+            # No more anyhow backtrace is printed.
+            assert msg.endswith("error: Cancelled\n")
```

Comment:
```
nextmsg, offset = wait_until(20, 0.5, lambda: end.pageserver.assert_log_contains(".*", offset))
```

the above should give us the next message, and if it would start with non-whitespace character (timestamp) then it would not be a debug formatted anyhow error. However, we don't really need to go there, because the variant now has no anyhow in it all, so I think we could remove this line as well.